frontend/packages/dev-console/src/components/import/GitImport.tsx

```diff
@@ -84,26 +78,15 @@ const GitImport: React.FC<GitImportProps> = ({ namespace, imageStreams }) => {
 
     const {
       project: { name: projectName },
-      route: { create: canCreateRoute },
-      image: { ports },
     } = values;
 
-    const requests = [
-      createDeploymentConfig(values, imageStream),
-      createImageStream(values, imageStream),
-      createBuildConfig(values, imageStream),
-    ];
+    const dryRunRequests: Promise<K8sResourceKind>[] = createResources(values, imageStream, true);
 
-    // Only create a service or route if the builder image has ports.
-    if (!_.isEmpty(ports)) {
-      requests.push(createService(values, imageStream));
-      if (canCreateRoute) {
-        requests.push(createRoute(values, imageStream));
-      }
-    }
-
-    requests.forEach((r) => r.catch((err) => actions.setStatus({ submitError: err.message })));
-    Promise.all(requests)
+    Promise.all(dryRunRequests)
+      .then(() => {
+        const requests: Promise<K8sResourceKind>[] = createResources(values, imageStream, false);
+        return Promise.all(requests);
+      })
```

Comment:
If `createResources` returned `Promise.all`, wouldn't this simply become:
```
createResources(values, imageStream, true)
    .then(() => createResources(values, imageStream))
    .then(...)
```

Reply:
Simplified.